src/plugins/data_source_management/public/components/utils.ts

```diff
@@ -123,11 +123,14 @@ export const fetchDataSourceConnections = async (
   hideLocalCluster: boolean
 ) => {
   try {
-    const directQueryConnectionsPromises = dataSources.map((ds) =>
-      getDirectQueryConnections(ds.id, http!).catch(() => [])
-    );
-    const directQueryConnectionsResult = await Promise.all(directQueryConnectionsPromises);
-    const directQueryConnections = directQueryConnectionsResult.flat();
+    const directQueryConnections: DataSourceTableItem[] = [];
+
+    for await (const datasource of dataSources) {
+      await getDirectQueryConnections(datasource.id, http!)
+        .then((connections) => directQueryConnections.push(...connections))
+        .catch(() => directQueryConnections.push([]));
```

Comment:
before, empty arrays were flattened out - is this handled downstream?

Reply:
actually I just realized by applying correct timeout alone can fix the issue. If Promise timeout faster, it won't pile up memory with Promise.all and cause memory issue. I just reverted the change to keep original Promise.all() for sake of performance. Also, 5s timeout for the specific calls on DSM table page is reasonable, as it's just a call to verify if the endpoint is reachable, no need to wait for the default 30s timeout to call with a single datasource endpoint
cc: @joshuali925 @huyaboo